clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusTargetGroupServerGroupProvider.java

```diff
@@ -133,4 +125,75 @@ public TitusTargetGroupServerGroupProvider(Cache cacheView) {
     return source.getRelationships().get(relationship) != null ? cacheView.getAll(relationship, source.getRelationships().get(relationship)) : Collections.emptyList();
   }
 
+  private Optional<LoadBalancerInstance> getInstanceHealth(String instanceKey,
+                                                          Map instanceDetails,
+                                                          Map<String, String> targetGroupDetails) {
+    String healthKey;
+    try {
+      healthKey = com.netflix.spinnaker.clouddriver.aws.data.Keys.getInstanceHealthKey(
+        ((Map) instanceDetails.get("task")).get("containerIp").toString(),
+        targetGroupDetails.get("account"),
+        targetGroupDetails.get("region"),
+        "aws-load-balancer-v2-target-group-instance-health"
+      );
+    } catch (NullPointerException e) {
+      return Optional.empty();
+    }
+
+    CacheData healthData = cacheView.get(HEALTH.ns, healthKey);
+
+    Map<String, Object> health = getTargetGroupHealth(instanceKey, targetGroupDetails, healthData);
+
+    return Optional.of(new LoadBalancerInstance(
+      ((Map) instanceDetails.get("task")).get("id").toString(),
+      null,
+      health
+    ));
+  }
+
+  private static Map<String, Object> getTargetGroupHealth(String instanceKey,
+                                                   Map<String, String> targetGroupDetails,
+                                                   CacheData healthData) {
+    try {
+      if (healthDataContainsTargetGroups(healthData)) {
+        Map targetGroupHealth = getTargetGroupHealthData(targetGroupDetails, healthData);
+
+        if (!targetGroupHealth.isEmpty()) {
+          Map<String, Object> health = new HashMap<>();
```

Comment:
Also a small bit of efficiency here. Doing a static initialization of a hashmap `e.g. new HashMap<>() { /* put commands */ };` creates a temporary class and doesn't buy us much terseness anyhow.